### PR TITLE
Test the main game directory's mod subdirectory for write permissions

### PR DESCRIPTION
**Pull request by penev92**
And add the result to the system information query.

To (eventually) settle the argument that took place on IRC tonight.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs [hunk 1]
```diff
@@ -266,7 +266,21 @@ string SysInfoQuery()
...
-				Uri.EscapeUriString(Game.ModData.Manifest.Mod.Version));
+				Uri.EscapeUriString(Game.ModData.Manifest.Mod.Version),
+				Uri.EscapeUriString(TestWritePermissions().ToString()));
+		}
+
+		byte TestWritePermissions()
+		{
+			try
+			{
+				using (var fs = File.Create(Path.Combine(".", "mods", ".testwritable"), 1, FileOptions.DeleteOnClose))
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs]: To settle the argument you will also need to show that Windows isn't redirecting the write to the VirtualStore directory.

**Comment by GraionDilach** [comment]
:+1:

This will prove me valid in the long term anyway.

**Comment by pchote** [comment]
Note to others: The context of this argument is that @penev92 and @GraionDilach strongly believe that we should expose UI allowing users to install content (mods/maps) into the OpenRA install directory because there is a subset of users who either run the game with admin privileges, or who don't know / are happy to rely on Windows's VirtualStore write redirection.

**Comment by GraionDilach** [comment]
My point is that most players on Windows will install their game to another partition and not to the default Program Files because it's a game and not an application.

**Comment by penev92** [comment]
Let's stay on topic. This is strictly for collecting information **on the basis of which** we can later argue.

Also allowing that might be terrible idea, but no worse than what we do know.

File: OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs (hunk 1 above)
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs]: Why not use `Directory.GetAccessControl`?

**Comment by Mailaender** [comment]
I assume the Windows VirtualStore write redirection is a workaround for legacy applications that don't have the ability to put configuration files at proper user directories. It is not quite clear to me what this is trying to audit and why we need it. However a check and a user friendly message like "don't run OpenRA with administrator rights" or "C:\ProgramFiles\OpenRA\Support\ isn't writable" might help us have less obvious and repetitive support cases where people confuse this with a 1995 era game. It seems what Windows users try when things don't work is reinstall, run as admin or compatibility mode when instead they should check their graphics drivers or other common problems from the FAQ.

**Comment by obrakmann** [comment]
I think the discussion here is missing the point a bit. Graion's intention is to not install mods onto his (smaller, presumably) C: partition, where the user's support directory would be located by default. His solution is to allow installing mods into the game directory, whose location the user can customize during installation. Of course, as mentioned by pchote, that solution is horrible. Users shouldn't mess with the main installation.

The proper solution, in my eyes, would be to allow the user to specify the location of the user's support directory as well. This is currently not possible, afaik. Then people would be able to move their stuff from the C: drive to whatever other drive they wish.

**Comment by pchote** [comment]
I'd be all for replacing the hardcoded game-dir support dir override with a more general commandline option to specify the support dir.  People on windows who want a portable or custom install can then edit their game shortcut (or we could even have the game installer write custom shortcuts) to use it.

**Comment by obrakmann** [comment]
After thinking some more about it, an even better solution would be to split the support dir into a "Settings" and a "Content" directory. The former would contain settings.yaml, news.yaml, motd.txt and the Logs directory and be located in the same place where the Support dir currently is. The content directory would contain the Content, mods, maps and Replays directories. Its default location could be the same place as the settings directory, or somewhere else, TBD.

The advantage would be that we can just use a normal game settings property that gets saved to settings.yaml to change the support directory and won't have to do any gymnastics with command-line parameters and editing shortcuts. Also something similar was already planned for Linux: #10374.

**Comment by GraionDilach** [comment]
>I'd be all for replacing the hardcoded game-dir support dir override with a more general commandline option to specify the support dir. People on windows who want a portable or custom install can then edit their game shortcut (or we could even have the game installer write custom shortcuts) to use it.

This makes the modchooser redundant and switching mods on the MP screen a mess.

However specifying additional such custom Content folders and then listing them in the settings.yaml sounds like a solution. The game could drop the invalidated ones during launch then.

**Comment by pchote** [comment]
The ability to specify the location of the support directory is completely unrelated to the mod chooser.

**Comment by GraionDilach** [comment]
It isn't. Specifying a custom Content folder for mods with shortcut B while having all the other mod contents in default location under shortcut A will crash the game if you want to switch from one affected mod of shortcut A to one under shortcut B, and will also mess up the remote MP launching as well, since you cannot rely on command-line arguments on that.

**Comment by obrakmann** [comment]
There would only be one Content folder, not additional ones.

**Comment by pchote** [comment]
The support directory is a single location that is used for *all* user-created / user-installed content (settings, log files, maps, replays, mods).  It doesn't make sense to talk about different directories for different mods.

**Comment by pchote** [comment]
> mess up the remote MP launching as well, since you cannot rely on command-line arguments on that.

Under windows the support dir location chosen during installation could be stored in the registry, and then the command-line argument would be provided by the windows-specific launcher.

**Comment by Phrohdoh** [comment]
If this is merged are the people for it going to be ok with its removal after significant data is gathered?
I see no reason to keep this around in the long-term.

**Comment by LipkeGu** [comment]
We could implement an option in the Installer for specifying the Location of the Content directory :) point it by default to the Documents Directory ...

**Comment by HenrytheSlav** [comment]
@GraionDilach 
> My point is that most players on Windows will install their game to another partition and not to the > default Program Files because it's a game and not an application.

Sorry, but almost all of users I met install everything into Program Files because they either believe that this is the correct folder to contain, well, program files, or don't have another partition (notebooks). If we are talking common users here of course.

**Comment by reaperrr** [comment]
Apart from some discussion there hasn't been any progress in quite a while, so I'm closing this for now to clear the PR queue. @penev92 just reopen when you get back to this.

**Pull request closed without merging**